configuration/configuration_blackbox_test.go

```diff
@@ -0,0 +1,108 @@
+package configuration
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"net/http"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/goadesign/goa"
+	"github.com/stretchr/testify/assert"
+)
+
+var req *goa.RequestData
+
+func TestMain(m *testing.M) {
+	resetConfiguration()
+
+	req = &goa.RequestData{
+		Request: &http.Request{Host: "api.service.domain.org"},
+	}
+	os.Exit(m.Run())
+}
+
+func resetConfiguration() {
+	if err := Setup(""); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+}
+
+func TestGetKeycloakEndpointAuthDevModeOK(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	url, err := GetKeycloakEndpointAuth(req)
+	assert.Nil(t, err)
+	// In dev mode it's always the defualt value regardless of the request
+	assert.Equal(t, "http://sso.demo.almighty.io/auth/realms/demo/protocol/openid-connect/auth", url)
+}
+
+func TestGetKeycloakEndpointAuthSetByEnvVaribaleOK(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	env := os.Getenv("ALMIGHTY_KEYCLOAK_ENDPOINT_AUTH")
+	defer func() {
+		os.Setenv("ALMIGHTY_KEYCLOAK_ENDPOINT_AUTH", env)
+		resetConfiguration()
+	}()
+
+	os.Setenv("ALMIGHTY_KEYCLOAK_ENDPOINT_AUTH", "authEndpoint")
+	resetConfiguration()
+
+	url, err := GetKeycloakEndpointAuth(req)
+	assert.Nil(t, err)
+	assert.Equal(t, "authEndpoint", url)
```

Comment:
As per my previous feedback, the order of preference of endpoint(s) should be :
1. dev mode defaults if dev mode is enabled. ( will never be in prod )
2. env variables if set. ( may or may not be present in prod )
3. request ( always present in production )

In that case, this test should 'unset' dev mode temporarily and then test for env var as you are doing now..